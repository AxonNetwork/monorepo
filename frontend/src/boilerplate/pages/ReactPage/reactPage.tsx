import * as React from 'react';
import { InjectedI18nProps, InjectedTranslateProps, translate } from 'react-i18next';

import FetchNote from './components/FetchNote';
import TodoLayout from './components/TodoLayout';
import i18ns from './i18n';

interface IReactPageProps extends InjectedI18nProps, InjectedTranslateProps { }

export class ReactPage extends React.Component<IReactPageProps> {
  constructor(props: IReactPageProps) {
    super(props);
    this.loadI18ns();
  }

  public loadI18ns() {
    const { i18n } = this.props;
    Object.keys(i18ns).forEach((key) => {
      i18n.addResourceBundle(key, 'ReactPage', i18ns[key]);
    });
  }

  public render() {
    const { t } = this.props;
    return (
      <div>
        <div className='react-container'>
          <h1 className='page-title'>{t('title')}</h1>
          <TodoLayout />
          <FetchNote />
        </div>
      </div>
    );
  }
}

export default translate('ReactPage')(ReactPage);
