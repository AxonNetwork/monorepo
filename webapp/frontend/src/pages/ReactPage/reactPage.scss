@import 'sass/variables';

.react-block {
  border: 1px lightgray solid;
  margin: 50px auto;
  width: 850px;
  padding: 20px 30px 40px;
}

@media only screen and (max-width: $medium-screen) {
  .react-block {
    max-width: 80%;
  }
}

@media only screen and (max-width: $small-screen) {
  .react-block {
    max-width: 85%;
  }
}

.react-block-title {
  display: block;
  font-weight: 100;
  font-size: 4.5em;
  color: $secondary-color;
  margin-bottom: 20px;
}

@media only screen and (max-width: $medium-screen) {
  .react-block-title {
    font-weight: 100;
    font-size: 4em;
  }
}

@media only screen and (max-width: $small-screen) {
  .react-block-title {
    font-weight: 100;
    font-size: 3em;
  }
}
