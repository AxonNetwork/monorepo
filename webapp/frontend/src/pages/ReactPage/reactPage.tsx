import React from 'react';
import { withNamespaces, WithNamespaces as Ii18nProps } from 'react-i18next';

import FetchNote from './components/FetchNote';
import TodoLayout from './components/TodoLayout';
import i18ns from './i18n';

export class ReactPage extends React.Component<Ii18nProps> {
  constructor(props: Ii18nProps) {
    super(props);
    this.loadI18ns();
  }

  public loadI18ns() {
    const { i18n } = this.props;
    Object.keys(i18ns).forEach((key) => {
      i18n.addResourceBundle(key, 'ReactPage', i18ns[key]);
    });
  }

  public render() {
    const { t } = this.props;
    return (
      <div>
        <div className='react-container'>
          <h1 className='page-title'>{t('title')}</h1>
          <TodoLayout />
          <FetchNote />
        </div>
      </div>
    );
  }
}

export default withNamespaces('ReactPage')(ReactPage);
