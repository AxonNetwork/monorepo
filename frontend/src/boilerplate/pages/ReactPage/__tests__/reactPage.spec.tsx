import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';

import { ReactPage } from '../reactPage';

jest.mock('../components/FetchNote', () => ({ default: 'FetchNote' }));
jest.mock('../components/TodoLayout', () => ({ default: 'TodoLayout' }));
const tMock = (k: string) => k;
const i18nMock: any = {
  addResourceBundle: () => { return; },
};

describe('ReactPage', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <ReactPage t={tMock} i18n={i18nMock} />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
