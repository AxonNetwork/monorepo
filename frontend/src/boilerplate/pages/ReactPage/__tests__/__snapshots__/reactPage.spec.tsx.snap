// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactPage should renders correctly 1`] = `
<div>
  <div
    className="react-container"
  >
    <h1
      className="page-title"
    >
      title
    </h1>
    <TodoLayout />
    <FetchNote />
  </div>
</div>
`;
